checkstyle.xml

```diff
@@ -0,0 +1,127 @@
+    		<property name="severity" value="warning" />
+    	</module>
+    	<module name="MissingJavadocMethod">
+      		<property name="scope" value="public"/>
+      		<property name="allowedAnnotations" value="Override, Test, Deprecated"/>
+      		<property name="allowMissingPropertyJavadoc" value="true"/>
+      		<property name="tokens" value="METHOD_DEF"/>
+      		<property name="severity" value="warning" />
+    	</module>
+    	<module name="ArrayTypeStyle"/>
+    	<module name="CommentsIndentation"/>
+    	<module name="Indentation">
+    		<property name="forceStrictCondition" value="true" />
+    		<property name="severity" value="warning" />
+    	</module>
+    	<module name="OuterTypeFilename"/>
+    	<module name="ConstantName">
+  			<property name="format" value="^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>
+		</module>
+		<module name="ClassTypeParameterName">
+      		<property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
+    	</module>
+    	<module name="MethodTypeParameterName">
+      		<property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
+        </module>
+    	<module name="InterfaceTypeParameterName">
+      		<property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
+    	</module>
+    	<module name="LambdaParameterName"/>
+    	<module name="LocalVariableName">
+  			<property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/>
+		</module>
+		<module name="MemberName"/>
+		<module name="MethodName">
+   			<property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/>
+   			<property name="allowClassName" value="false"/>
+		</module>
+		<module name="ParameterName">
+  			<property name="format" value="^[a-z][_a-zA-Z0-9]+$"/>
+  			<property name="ignoreOverridden" value="true"/>
+		</module>
+		<module name="StaticVariableName">
+  			<property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/>
+  		</module>
+  		<module name="EmptyLineSeparator">
+      		<property name="allowNoEmptyLineBetweenFields" value="true"/>
+    	</module>
+    	<module name="GenericWhitespace"/>
+    	<module name="MethodParamPad"/>
+    	<module name="NoWhitespaceBefore">
+  			<property name="allowLineBreaks" value="true"/>
+		</module>
+		<module name="ParenPad"/>
+		<module name="WhitespaceAround">
+  			<property name="tokens" value="ASSIGN, DIV_ASSIGN,PLUS_ASSIGN, MINUS_ASSIGN,
+  			                               STAR_ASSIGN, MOD_ASSIGN, SR_ASSIGN, BSR_ASSIGN,
+  			                               SL_ASSIGN, BXOR_ASSIGN, BOR_ASSIGN, BAND_ASSIGN"/>
+		</module>
+    </module>
+    <module name="NewlineAtEndOfFile"/>
+    <module name="OrderedProperties"/>
+    <module name="FileLength">
+  		<property name="max" value="5000"/>
+	</module>
+	<module name="LineLength">
+    	<property name="fileExtensions" value="java"/>
+    	<property name="max" value="120"/>
+  	</module>
+  	<module name="FileTabCharacter">
+    	<property name="eachLine" value="true"/>
+  	</module>

```

Comment:
I would delete this module. FileTabCharacter checks the tab occurence in the code, so even pure SpringBoot App cause an error. We all use IntelliJ, so this is not so important.

Reply:
From my point of view, it's important. Tabs in our IDE's might correspond to diffrent number of spaces (as default), so it's better for formatting to use only spaces instead of Tabs (Tabs can be changed to proper value of spaces by IDE). I'm already not using InelliJ, so I can tell that your formatting is diffrent than mine in this case and I think it's better to set it in the IDE, everyone would have the same code formatting and problem would be solved.